File: solutions/haskell/isogram/1/src/Isogram.hs
module Isogram (isIsogram) where

import Data.Char
import Data.List

isIsogram :: String -> Bool
isIsogram s = all (\l -> length l == 1) $ group $ sort $ map toLower $ filter (\c -> c /= ' ' && c /= '-' && c /= '_') s

File: solutions/zig/isogram/1/isogram.zig
const std = @import("std");

pub fn isIsogram(str: []const u8) bool {
    for (0..str.len) |i| {
        if (std.ascii.isAlphabetic(str[i]) and i > 0) {
            for (0..i) |j| {
                if (std.ascii.toLower(str[i]) == std.ascii.toLower(str[j])) {
                    return false;
                }
            }
        }
    }
    return true;
}
